### PR TITLE
Fix RelativeExpiration with SlidingExpiration

### PR DESCRIPTION
**Pull request by BrennanConroy**
#194 
@Tratcher

### CODE REVIEW

File: src/Microsoft.Extensions.Caching.Redis/RedisCache.cs [hunk 1]
```diff
@@ -78,10 +78,16 @@ public void Set(string key, byte[] value, DistributedCacheEntryOptions options)
 
             var creationTime = DateTimeOffset.UtcNow;
 
+            var absoluteExpiration = options.AbsoluteExpiration;
+            if (options.AbsoluteExpirationRelativeToNow.HasValue)
+            {
+                absoluteExpiration = creationTime + options.AbsoluteExpirationRelativeToNow;
```
Comment (Tratcher) [inline src/Microsoft.Extensions.Caching.Redis/RedisCache.cs]: Should this absoluteExpiration be passed into GetExpirationInSeconds to reduce redundant logic there?

Comment (BrennanConroy) [inline src/Microsoft.Extensions.Caching.Redis/RedisCache.cs]: Yeah, I was thinking something like that

**Comment by BrennanConroy** [comment]
🆙 📅

**Comment by Tratcher** [comment]
:shipit:

**Pull request merged**